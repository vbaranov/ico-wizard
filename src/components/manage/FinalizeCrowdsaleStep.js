import React from 'react'
import { Link } from 'react-router-dom'
import classNames from 'classnames'

export const FinalizeCrowdsaleStep = ({ disabled, handleClick }) => (
  <div className="steps-content container">
    <div className="about-step">
      <div className="swal2-icon swal2-info warning-logo">!</div>
      <p className="title">Finalize Crowdsale</p>
      <p className="description">
        Finalize - Finalization is the last step of the crowdsale.
        You can make it only after the end of the last tier. After finalization, it's not possible to update tiers, buy
        tokens. All tokens will be movable, reserved tokens will be issued.
      </p>
      <Link to='#' onClick={() => !disabled ? handleClick() : undefined}>
      <span className={classNames(
        'button',
        {
          'button_disabled': disabled,
          'button_fill': !disabled
        }
      )}>
        Finalize Crowdsale
      </span>
      </Link>
    </div>
  </div>
)
