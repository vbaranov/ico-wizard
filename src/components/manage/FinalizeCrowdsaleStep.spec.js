import React from 'react'
import { StaticRouter } from 'react-router'
import { FinalizeCrowdsaleStep } from './FinalizeCrowdsaleStep'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'

configure({ adapter: new Adapter() })

describe('FinalizeCrowdsaleStep', () => {
  it('should render the component with active button', () => {
    const finalizeCrowdsaleStateParams = {
      disabled: false,
      handleClick: jest.fn()
    }

    expect(renderer.create(
      <StaticRouter location="testLocation" context={{}}>
        <FinalizeCrowdsaleStep {...finalizeCrowdsaleStateParams} />
      </StaticRouter>
    ).toJSON()).toMatchSnapshot()
  })

  it('should render the component with disabled button', () => {
    const finalizeCrowdsaleStateParams = {
      disabled: true,
      handleClick: jest.fn()
    }

    expect(renderer.create(
      <StaticRouter location="testLocation" context={{}}>
        <FinalizeCrowdsaleStep {...finalizeCrowdsaleStateParams} />
      </StaticRouter>
    ).toJSON()).toMatchSnapshot()
  })

  it('should call handleClick', () => {
    const finalizeCrowdsaleStateParams = {
      disabled: false,
      handleClick: jest.fn()
    }

    const wrapper = mount(
      <StaticRouter location="testLocation" context={{}}>
        <FinalizeCrowdsaleStep {...finalizeCrowdsaleStateParams} />
      </StaticRouter>
    )

    const button = wrapper.find('Link').at(0)

    button.simulate('click')

    expect(finalizeCrowdsaleStateParams.handleClick).toHaveBeenCalled()
  })

  it('should not call handleClick', () => {
    const finalizeCrowdsaleStateParams = {
      disabled: true,
      handleClick: jest.fn()
    }

    const wrapper = mount(
      <StaticRouter location="testLocation" context={{}}>
        <FinalizeCrowdsaleStep {...finalizeCrowdsaleStateParams} />
      </StaticRouter>
    )

    const button = wrapper.find('Link').at(0)

    button.simulate('click')

    expect(finalizeCrowdsaleStateParams.handleClick).toHaveBeenCalledTimes(0)
  })
})
