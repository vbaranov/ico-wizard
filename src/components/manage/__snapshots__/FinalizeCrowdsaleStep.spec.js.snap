// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalizeCrowdsaleStep should render the component with active button 1`] = `
<div
  className="steps-content container"
>
  <div
    className="about-step"
  >
    <div
      className="swal2-icon swal2-info warning-logo"
    >
      !
    </div>
    <p
      className="title"
    >
      Finalize Crowdsale
    </p>
    <p
      className="description"
    >
      Finalize - Finalization is the last step of the crowdsale. You can make it only after the end of the last tier. After finalization, it's not possible to update tiers, buy tokens. All tokens will be movable, reserved tokens will be issued.
    </p>
    <a
      href="/#"
      onClick={[Function]}
    >
      <span
        className="button button_fill"
      >
        Finalize Crowdsale
      </span>
    </a>
  </div>
</div>
`;

exports[`FinalizeCrowdsaleStep should render the component with disabled button 1`] = `
<div
  className="steps-content container"
>
  <div
    className="about-step"
  >
    <div
      className="swal2-icon swal2-info warning-logo"
    >
      !
    </div>
    <p
      className="title"
    >
      Finalize Crowdsale
    </p>
    <p
      className="description"
    >
      Finalize - Finalization is the last step of the crowdsale. You can make it only after the end of the last tier. After finalization, it's not possible to update tiers, buy tokens. All tokens will be movable, reserved tokens will be issued.
    </p>
    <a
      href="/#"
      onClick={[Function]}
    >
      <span
        className="button button_disabled"
      >
        Finalize Crowdsale
      </span>
    </a>
  </div>
</div>
`;
